### PR TITLE
golang: Verify if a session exists before adding

### PR DESCRIPTION
**Pull request by mestery**
This adds some logic around verifying a session does not already exist before adding it in.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: openoffload/go/server/sessionoffload.go [hunk 1]
```diff
@@ -127,54 +168,72 @@ func next_session_id() (uint64, error) {
...
+			}
+			resp.ResponseError = append(resp.ResponseError, &session_resp_err)
+			add_new_session = false
+		}
+
 		newSessionId, err := next_session_id()
 		if err != nil {
 			log.Printf("Error getting new session ID: %v", err)
-			return err
+			add_new_session = false
```
Comment (seroyer) [inline openoffload/go/server/sessionoffload.go:202]: What's the exit strategy here?  It won't add the session, but then will loop back and try again.  In this case, what's the expected behavior of the stream.Recv above on the next loop?

File: openoffload/go/server/sessionoffload.go [hunk 2]
```diff
@@ -93,6 +94,46 @@ func session_update() {
 	}
 }
 
+func find_session_by_7_tuple(in_lif, out_lif int32, source_ipv4 uint32, source_ipv6 []byte,
+			dest_ipv4 uint32, dest_ipv6 []byte,
+			src_port, dst_port uint32,
+			protocol_id *fw.ProtocolId, ip_version *fw.IpVersion) uint64 {
+	for _, v := range sessions {
```
[the remark this answers is not in the document]
Comment (mestery) [reply openoffload/go/server/sessionoffload.go:101]: Yes, for sure.

File: openoffload/go/server/sessionoffload.go (hunk 1 above)
Comment (mestery) [reply openoffload/go/server/sessionoffload.go:202]: This is handling a stream request, so we iterate on each request, and for the ones which fail (and for which we can get a session ID), we return an error. However, in this case, the fact that `next_session_id()` returns zero means the session table is full. So we should also append a `SessionResponseError` indicating that here. I'll push a commit to address this now. Great find Stephen!

**Comment by mestery** [comment]
Weird, I had to close and then re-open this as it was not taking the second commit.

**Review by seroyer — APPROVED** [review]
Awesome!

**Pull request merged**